cloud-service-providers/google-cloud/vertexai/python/nim-vertexai.ipynb

```diff
@@ -0,0 +1,1372 @@
+    },
+    "executionInfo": {
+     "elapsed": 2,
+     "status": "ok",
+     "timestamp": 1721238618341,
+     "user": {
+      "displayName": "",
+      "userId": ""
+     },
+     "user_tz": 240
+    },
+    "id": "5ieTra-OYt66",
+    "outputId": "8bc00099-f31f-423c-f7ae-3154c73b9697",
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import sys\n",
+    "\n",
+    "if \"google.colab\" in sys.modules:\n",
+    "\n",
+    "    from google.colab import auth\n",
+    "\n",
+    "    auth.authenticate_user()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b8ed8ff4-2deb-45af-ad9f-ccbd7d801a8f",
+   "metadata": {},
+   "source": [
+    "### Set Up\n",
+    "\n",
+    "The example provided is `llama3-8b-instruct` NIM, on Vertex AI Workbench Notebook `g2-standard-24` instance with NVIDIA L4 GPU.\n",
+    "\n",
+    "IAM role requirements:\n",
+    "* Artifact Registry Repository Administrator `(roles/artifactregistry.repoAdmin)` \n",
+    "* Storage Admin `(roles/storage.admin)`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "vn6KU7wfjew5",
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "executionInfo": {
+     "elapsed": 970,
+     "status": "ok",
+     "timestamp": 1721238645640,
+     "user": {
+      "displayName": "",
+      "userId": ""
+     },
+     "user_tz": 240
+    },
+    "id": "vn6KU7wfjew5",
+    "outputId": "ca54be92-1d89-43cf-96e8-25d28f7577b2",
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# Get account name\n",
+    "import requests\n",
+    "gcloud_token = !gcloud auth print-access-token\n",
+    "gcloud_tokeninfo = requests.get('https://www.googleapis.com/oauth2/v3/tokeninfo?access_token=' + gcloud_token[0]).json()\n",
+    "account_email = gcloud_tokeninfo['email']\n",
+    "account_name = gcloud_tokeninfo['email'].split('@')[0]\n",
+    "print(account_email)\n",
+    "print(account_name)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "g7chyrzCF9yoWk9IFyfdOe4W",
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/",
+     "height": 35
+    },
+    "executionInfo": {
+     "elapsed": 196,
+     "status": "ok",
+     "timestamp": 1721251321866,
+     "user": {
+      "displayName": "",
+      "userId": ""
+     },
+     "user_tz": 240
+    },
+    "id": "g7chyrzCF9yoWk9IFyfdOe4W",
+    "outputId": "e2090fb2-bca8-46c9-c79b-d2369a53377f",
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# NIM: llama3-8b-instruct\n",
+    "region = \"us-central1\" # please set here\n",
+    "project_id = None # please set here\n",
+    "public_repository = None # please set here\n",
+    "private_repository = account_name\n",
+    "bucket_url = f\"gs://{account_name}\"\n",
+    "\n",
+    "nim_model = \"nim:llama3-8b-instruct-latest\"\n",
+    "# NIM in NGC\n",
+    "ngc_nim_image = \"nvcr.io/nim/meta/llama3-8b-instruct:latest\"\n",
```

Comment:
Should try not to use latest tag if possible

Reply:
Fixed. Thank you.